### PR TITLE
Replace crashlytics upload script with app distribution; Disable upload

### LINKED ISSUE #544: Project Setup: Build system & gradle
- [x] Define new build pipeline for safe v2
- [ ] Add/Adjust gradle files: lib templates, shared dependencies, test dependencies

### PR DESCRIPTION
**Pull request by elgatovital**
Closes #544 .

Changes proposed in this pull request:
- Adjust pipeline
- Disable app distribution upload for now
- Clean gradle setup

@gnosis/mobile-devs

### CODE REVIEW

File: ci/upload_app_distribution.sh [hunk 1]
```diff
@@ -5,12 +5,14 @@ set -e
...
 if [[ $BUILKITE_BRANCH  =~ ^v[0-9]+.* ]]; then
     export APP_VERSION_NAME=${BUILDKITE_BRANCH:1}
 fi
 
 echo "INFURA_API_KEY=$INFURA_API_KEY" > project_keys
 
-./gradlew assembleInternal assembleRinkeby assembleRelease
-./gradlew crashlyticsUploadDistributionInternal crashlyticsUploadDistributionRinkeby crashlyticsUploadDistributionRelease
+./gradlew assembleInternal
+./gradlew appDistributionUploadInternal
```
Comment (jpalvarezl) [inline ci/upload_app_distribution.sh:18]: Does this also require setup for app distribution in order to restore rinkeby and release? If that is the case maybe a TODO here too would be nice.

File: buildsystem/android_library.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8
```
Comment (jpalvarezl) [inline buildsystem/android_library.gradle:42]: Hehehe I see you snuck these in anyway :D I think we were going to have to add them at some point anyway

File: buildsystem/test_dependencies.gradle [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    testImplementation "androidx.arch.core:core-testing:2.1.0"
+
+    testImplementation "org.jetbrains.kotlin:kotlin-reflect:${versions.kotlin}"
+    testImplementation "com.github.gnosis.svalinn-kotlin:utils-testing:${versions.svalinn}"
+    testImplementation "org.mockito:mockito-core:${versions.mockito}"
+    testImplementation "org.mockito:mockito-inline:${versions.mockito}"
+    testImplementation "org.powermock:powermock-module-junit4:${versions.powermock}"
+    testImplementation "org.powermock:powermock-api-mockito2:${versions.powermock}"
+    testImplementation "org.powermock:powermock-core:${versions.powermock}"
+    testImplementation "org.powermock:powermock-module-junit4-rule:${versions.powermock}"
```
Comment (jpalvarezl) [inline buildsystem/test_dependencies.gradle:17]: I doubt we need powermock and mockito having mockk. We can do statics with it too.

File: gradle.properties [hunk 4]
```diff
@@ -13,4 +13,3 @@ org.gradle.jvmargs=-Xmx8g
 # org.gradle.parallel=true
 android.enableJetifier=true
 android.useAndroidX=true
-android.enableR8 = true
```
Comment (jpalvarezl) [inline gradle.properties:16]: What issues was this causing?

File: app/build.gradle [hunk 5]
```diff
@@ -242,32 +243,9 @@ dependencies {
...
-    })
-    androidTestImplementation "androidx.test.ext:junit:$versions.androidx_test_ext"
-
-    androidTestImplementation "android.arch.persistence.room:testing:$versions.androidx_room"
-
 }
 
+
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.AbstractKotlinCompile).all {
     kotlinOptions.freeCompilerArgs += ["-Xuse-experimental=kotlinx.coroutines.ExperimentalCoroutinesApi"]
```
Comment (jpalvarezl) [inline app/build.gradle:250]: I have a feeling we can get rid of this too, but I haven't checked it myself yet

File: gradle.properties (hunk 4 above)
Comment (elgatovital) [reply gradle.properties:16]: its deprecated

File: buildsystem/android_library.gradle (hunk 2 above)
Comment (elgatovital) [reply buildsystem/android_library.gradle:42]: we had them in :app already, no illegal activity here ;)

File: buildsystem/test_dependencies.gradle (hunk 3 above)
Comment (elgatovital) [reply buildsystem/test_dependencies.gradle]: removed them for now

**Pull request merged**